Zend/zend_exceptions.stub.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+interface Throwable
+{
+    /** @return string */
+    function getMessage();
+
+    /** @return int */
+    function getCode();
+
+    /** @return string */
+    function getFile();
+
+    /** @return int */
+    function getLine();
+
+    /** @return array */
+    function getTrace();
+
+    /** @return ?Throwable */
+    function getPrevious();
+
+    /** @return string */
+    function getTraceAsString();
+
+    /** @return string */
+    function __toString();
+}
+
+class Exception implements Throwable
+{
+    final private function __clone() {}
+
+    function __construct(string $message = UNKNOWN, int $code = 0, ?Throwable $previous = null) {}
+
+    function __wakeup() {}
+
+    /** @return string */
+    final function getMessage() {}
+
+    /** @return int */
+    final function getCode() {}
+
+    /** @return string */
+    final function getFile() {}
+
+    /** @return int */
+    final function getLine() {}
+
+    /** @return array */
+    final function getTrace() {}
+
+    /** @return ?Throwable */
+    final function getPrevious() {}
+
+    /** @return string */
```

Comment:
Looks like this one can also return `false` :(

Reply:
Oh, indeed: https://github.com/php/php-src/blob/0d55456d7349f308a018672301fedab563ca2ccc/Zend/zend_exceptions.c#L633-L635
I wonder whether this can actually happen.

Reply:
Ah, that special casing has been introduced with a894a8155fab068d68a04bf181dbaddfa01ccbb0. Since this is about unserialization of crafted input, I wonder whether we could throw an exception here (or even error out).

Reply:
Yeah, throwing should be fine.

Reply:
Okay.